### PR TITLE
[ios_platform_images] Update minimum version to iOS 11

### PR DESCRIPTION
**Pull request by jmagman**
`ios_platform_images` part of https://github.com/flutter/flutter/issues/101965.

Built the `ios_platform_images` example projects on Flutter 3.3. and let the tool auto-migrate various Xcode project/Podfile files.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/ios_platform_images/pubspec.yaml [hunk 1]
```diff
@@ -2,10 +2,10 @@ name: ios_platform_images
 description: A plugin to share images between Flutter and iOS in add-to-app setups.
 repository: https://github.com/flutter/plugins/tree/main/packages/ios_platform_images
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+ios_platform_images%22
-version: 0.2.1
+version: 0.2.2
 
 environment:
-  sdk: ">=2.14.0 <3.0.0"
+  sdk: '>=2.18.0 <3.0.0'
```
Comment (jmagman) [inline packages/ios_platform_images/pubspec.yaml:8]: This should have been bumped with 0.2.1 when updated to 3.3.

File: packages/ios_platform_images/ios/ios_platform_images.podspec [hunk 2]
```diff
@@ -17,7 +17,7 @@ Downloaded by pub (not CocoaPods).
   s.documentation_url = 'https://pub.dev/packages/ios_platform_images'
   s.source_files = 'Classes/**/*'
   s.dependency 'Flutter'
-  s.platform = :ios, '9.0'
+  s.platform = :ios, '11.0'
```
Comment (jmagman) [inline packages/ios_platform_images/ios/ios_platform_images.podspec:20]: This was manually edited.

File: packages/ios_platform_images/pubspec.yaml (hunk 1 above)
Comment (stuartmorgan) [reply packages/ios_platform_images/pubspec.yaml:8]: We don't generally bother to keep them in sync. I've thought about having tooling enforce expected pairs, but I haven't come up with any benefit from doing so.

So the current approach is that we mostly only update each one when there is a specific reason for that constraint. Usually it's just the Flutter constraint, but things like NNBD or new Dart features trigger an `sdk` update. Or when people feel like it, like here :)

File: packages/ios_platform_images/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 0.2.2
+
+* Updates minimum version to iOS 11.
```
Comment (stuartmorgan) [inline packages/ios_platform_images/CHANGELOG.md:3]: The [README](https://github.com/flutter/plugins/blob/main/packages/ios_platform_images/README.md?plain=1#L13) needs an update too. We haven't built out tooling to enforce that it matches reality yet (https://github.com/flutter/flutter/issues/84200).

**Review by stuartmorgan — APPROVED** [review]
LGTM modulo README update

**Pull request merged**